src/core/utils/internal/parseGraphQLRequest.ts

```diff
@@ -167,39 +167,44 @@ async function getGraphQLInput(request: Request): Promise<GraphQLInput | null> {
   }
 }
 
+const parsedGraphQLRequestCache = new WeakMap<Request, ParsedGraphQLRequest>()
 /**
  * Determines if a given request can be considered a GraphQL request.
  * Does not parse the query and does not guarantee its validity.
  */
 export async function parseGraphQLRequest(
   request: Request,
 ): Promise<ParsedGraphQLRequest> {
-  const input = await getGraphQLInput(request)
+  if (!parsedGraphQLRequestCache.has(request)) {
+    const input = await getGraphQLInput(request)
 
-  if (!input || !input.query) {
-    return
-  }
-
-  const { query, variables } = input
-  const parsedResult = parseQuery(query)
-
-  if (parsedResult instanceof Error) {
-    const requestPublicUrl = getPublicUrlFromRequest(request)
+    if (!input || !input.query) {
+      parsedGraphQLRequestCache.set(request, undefined)
+      return
+    }
 
-    throw new Error(
-      devUtils.formatMessage(
-        'Failed to intercept a GraphQL request to "%s %s": cannot parse query. See the error message from the parser below.\n\n%s',
-        request.method,
-        requestPublicUrl,
-        parsedResult.message,
-      ),
-    )
-  }
+    const { query, variables } = input
+    const parsedResult = parseQuery(query)
+
+    if (parsedResult instanceof Error) {
+      const requestPublicUrl = getPublicUrlFromRequest(request)
+      parsedGraphQLRequestCache.set(request, undefined)
```

Comment:
this is the only functional difference, since we'll throw on the _first request only_ and `undefined` will be the result for other requests, which should avoid spamming the logs